--- origsrc/gnome-commander-1.2.8.8/configure.in	2010-09-08 16:41:13.000000000 -0500
+++ src/gnome-commander-1.2.8.8/configure.in	2010-09-21 00:23:53.700106900 -0500
@@ -29,6 +29,24 @@ AC_SUBST(CC_WARNINGS)
 
 
 dnl ===================
+dnl Win32
+dnl ===================
+
+case "$host_os" in
+cygwin*|mingw*)
+  platform_win32=yes
+  NO_UNDEFINED="-no-undefined -Wl,\$(top_builddir)/src/gnome-commander.exe.a"
+  ;;
+*)
+  platform_win32=no
+  NO_UNDEFINED=""
+  ;;
+esac
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(NO_UNDEFINED)
+
+
+dnl ===================
 dnl Gettext stuff
 dnl ===================
 
--- origsrc/gnome-commander-1.2.8.8/libgcmd/Makefile.am	2010-09-08 16:41:13.000000000 -0500
+++ src/gnome-commander-1.2.8.8/libgcmd/Makefile.am	2010-09-21 00:24:14.175226100 -0500
@@ -5,7 +5,7 @@ AM_CPPFLAGS = \
 	$(GNOMEUI_CFLAGS) \
 	$(GNOMEVFS_CFLAGS)
 
-pkglib_LTLIBRARIES = libgcmd.la
+noinst_LTLIBRARIES = libgcmd.la
 
 libgcmd_la_SOURCES = \
 	gnome-cmd-plugin.h gnome-cmd-plugin.cc \
--- origsrc/gnome-commander-1.2.8.8/plugins/cvs/Makefile.am	2010-09-08 16:41:13.000000000 -0500
+++ src/gnome-commander-1.2.8.8/plugins/cvs/Makefile.am	2010-09-21 00:19:48.929922300 -0500
@@ -16,4 +16,4 @@ libcvs_la_SOURCES = \
 	interface.h interface.cc \
 	cvs-plugin.xpm close.xpm
 
-libcvs_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version
+libcvs_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version $(NO_UNDEFINED)
--- origsrc/gnome-commander-1.2.8.8/plugins/fileroller/Makefile.am	2010-09-08 16:41:13.000000000 -0500
+++ src/gnome-commander-1.2.8.8/plugins/fileroller/Makefile.am	2010-09-21 00:19:59.763718600 -0500
@@ -14,4 +14,4 @@ libfileroller_la_SOURCES = \
 	file-roller-plugin.h file-roller-plugin.cc \
 	file-roller.xpm file-roller-small.xpm
 
-libfileroller_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version
+libfileroller_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version $(NO_UNDEFINED)
--- origsrc/gnome-commander-1.2.8.8/plugins/test/Makefile.am	2010-09-08 16:41:13.000000000 -0500
+++ src/gnome-commander-1.2.8.8/plugins/test/Makefile.am	2010-09-21 00:20:04.020151700 -0500
@@ -14,4 +14,4 @@ libtest_la_SOURCES = \
 	test-plugin.h test-plugin.cc \
 	test-plugin.xpm
 
-libtest_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version
+libtest_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version $(NO_UNDEFINED)
--- origsrc/gnome-commander-1.2.8.8/src/Makefile.am	2010-09-08 16:41:13.000000000 -0500
+++ src/gnome-commander-1.2.8.8/src/Makefile.am	2010-09-21 00:20:56.714027200 -0500
@@ -113,4 +113,10 @@ gnome_commander_LDADD = \
 	$(PYTHON_LIBS) \
 	$(PYTHON_EXTRA_LIBS)
 
+gnome_commander_LDFLAGS = -export-dynamic
+
+if PLATFORM_WIN32
+gnome_commander_LDFLAGS += -Wl,--out-implib,gnome-commander.exe.a
+endif
+
 gcmd_block_SOURCES = block.cc
