--- origsrc/gnome-commander-1.2.8/src/dialogs/Makefile.am	2009-06-29 04:57:46.000000000 -0500
+++ src/gnome-commander-1.2.8/src/dialogs/Makefile.am	2009-07-17 19:55:37.394642400 -0500
@@ -5,7 +5,7 @@ noinst_LIBRARIES = libgcmd-dialogs.a
 AM_CPPFLAGS = \
 	$(CC_WARNINGS) \
 	-I$(top_srcdir) \
-	-I.. \
+	-I$(top_srcdir)/src \
 	$(GNOMEUI_CFLAGS) \
 	$(GNOMEVFS_CFLAGS) \
 	$(EXIV2_CFLAGS) \
--- origsrc/gnome-commander-1.2.7/src/intviewer/Makefile.am	2008-01-03 16:41:10.000000000 -0600
+++ src/gnome-commander-1.2.7/src/intviewer/Makefile.am	2008-07-28 17:13:06.390625000 -0500
@@ -3,7 +3,7 @@
 AM_CPPFLAGS = \
 	$(CC_WARNINGS) \
 	-I$(top_srcdir) \
-	-I.. \
+	-I$(top_srcdir)/src \
 	$(GLIB_CFLAGS) \
 	$(GTK_CFLAGS) \
 	$(GNOMEUI_CFLAGS) \
--- origsrc/gnome-commander-1.2.7/src/tags/Makefile.am	2007-11-21 16:43:12.000000000 -0600
+++ src/gnome-commander-1.2.7/src/tags/Makefile.am	2008-07-28 17:13:06.453125000 -0500
@@ -5,7 +5,7 @@
 AM_CPPFLAGS = \
 	$(CC_WARNINGS) \
 	-I$(top_srcdir) \
-	-I.. \
+	-I$(top_srcdir)/src \
 	$(GNOMEUI_CFLAGS) \
 	$(GNOMEVFS_CFLAGS) \
 	$(EXIV2_CFLAGS) \
