--- origsrc/gnome-commander-1.2.8.6/configure.in	2010-06-01 14:07:14.000000000 -0500
+++ src/gnome-commander-1.2.8.6/configure.in	2010-06-25 02:47:41.525811400 -0500
@@ -27,6 +27,26 @@ if test "x$GCC" = "xyes"; then
 fi
 AC_SUBST(CC_WARNINGS)
 
+dnl ===================
+dnl Win32
+dnl ===================
+
+case "$host_os" in
+cygwin*|mingw*)
+  platform_win32=yes
+  privlibdir="${libdir}"
+  PLUGIN_LDFLAGS="-module -avoid-version -no-undefined -Wl,\$(top_builddir)/src/gnome-commander.exe.a"
+  ;;
+*)
+  platform_win32=no
+  privlibdir="${pkglibdir}"
+  PLUGIN_LDFLAGS="-module -avoid-version"
+  ;;
+esac
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(privlibdir)
+AC_SUBST(PLUGIN_LDFLAGS)
+
 
 dnl ===================
 dnl Gettext stuff
@@ -302,7 +322,7 @@ Py_Exit (0);
 }
 EOF
 
-    if ${CC} $PYTHON_CFLAGS $PYTHON_LIBS $PYTHON_EXTRA_LIBS -fPIC -shared -o testpython.so testpython.c -Wl,-soname,testpython.so >/dev/null 2>&1 && \
+    if ${CC} $PYTHON_CFLAGS -fPIC -shared -o testpython.so testpython.c -Wl,-soname,testpython.so $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
         ( objdump --headers --private-headers testpython.so | grep 'testpython' ) >/dev/null 2>&1; then
         result=yes
     else
--- origsrc/gnome-commander-1.2.8.6/libgcmd/Makefile.am	2010-03-04 14:06:22.000000000 -0600
+++ src/gnome-commander-1.2.8.6/libgcmd/Makefile.am	2010-06-25 02:47:41.529811600 -0500
@@ -5,7 +5,7 @@ AM_CPPFLAGS = \
 	$(GNOMEUI_CFLAGS) \
 	$(GNOMEVFS_CFLAGS)
 
-pkglib_LTLIBRARIES = libgcmd.la
+noinst_LTLIBRARIES = libgcmd.la
 
 libgcmd_la_SOURCES = \
 	gnome-cmd-plugin.h gnome-cmd-plugin.cc \
--- origsrc/gnome-commander-1.2.8.6/plugins/cvs/Makefile.am	2010-03-04 14:06:22.000000000 -0600
+++ src/gnome-commander-1.2.8.6/plugins/cvs/Makefile.am	2010-06-25 02:47:41.533811900 -0500
@@ -16,4 +16,5 @@ libcvs_la_SOURCES = \
 	interface.h interface.cc \
 	cvs-plugin.xpm close.xpm
 
-libcvs_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS)
+libcvs_la_LDFLAGS = $(PLUGIN_LDFLAGS)
+libcvs_la_LIBADD = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS)
--- origsrc/gnome-commander-1.2.8.6/plugins/fileroller/Makefile.am	2010-03-04 14:06:22.000000000 -0600
+++ src/gnome-commander-1.2.8.6/plugins/fileroller/Makefile.am	2010-06-25 02:47:41.537812100 -0500
@@ -14,4 +14,5 @@ libfileroller_la_SOURCES = \
 	file-roller-plugin.h file-roller-plugin.cc \
 	file-roller.xpm file-roller-small.xpm
 
-libfileroller_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS)
+libfileroller_la_LDFLAGS = $(PLUGIN_LDFLAGS)
+libfileroller_la_LIBADD = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS)
--- origsrc/gnome-commander-1.2.8.6/plugins/test/Makefile.am	2010-03-04 14:06:22.000000000 -0600
+++ src/gnome-commander-1.2.8.6/plugins/test/Makefile.am	2010-06-25 02:47:41.541812300 -0500
@@ -14,4 +14,5 @@ libtest_la_SOURCES = \
 	test-plugin.h test-plugin.cc \
 	test-plugin.xpm
 
-libtest_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS)
+libtest_la_LDFLAGS = $(PLUGIN_LDFLAGS)
+libtest_la_LIBADD = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS)
--- origsrc/gnome-commander-1.2.8.6/src/Makefile.am	2010-06-01 13:53:06.000000000 -0500
+++ src/gnome-commander-1.2.8.6/src/Makefile.am	2010-06-25 02:49:36.024360300 -0500
@@ -113,4 +113,10 @@ gnome_commander_LDADD = \
 	$(PYTHON_LIBS) \
 	$(PYTHON_EXTRA_LIBS)
 
+gnome_commander_LDFLAGS = -export-dynamic
+
+if PLATFORM_WIN32
+gnome_commander_LDFLAGS += -Wl,--out-implib,gnome-commander.exe.a
+endif
+
 gcmd_block_SOURCES = block.cc
