--- origsrc/gnome-commander-1.4.1/configure.ac	2014-04-05 02:38:11.000000000 -0500
+++ src/gnome-commander-1.4.1/configure.ac	2014-04-30 20:40:34.306555500 -0500
@@ -29,6 +29,24 @@ AC_SUBST(CC_WARNINGS)
 
 
 dnl ===================
+dnl Win32
+dnl ===================
+
+case "$host_os" in
+cygwin*|mingw*)
+  platform_win32=yes
+  NO_UNDEFINED="-no-undefined -Wl,\$(top_builddir)/src/gnome-commander.exe.a"
+  ;;
+*)
+  platform_win32=no
+  NO_UNDEFINED=""
+  ;;
+esac
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(NO_UNDEFINED)
+
+
+dnl ===================
 dnl Gettext stuff
 dnl ===================
 
@@ -328,7 +346,7 @@ Py_Exit (0);
 }
 EOF
 
-    if ${CC} $PYTHON_CFLAGS $PYTHON_LIBS $PYTHON_EXTRA_LIBS -fPIC -shared -o testpython.so testpython.c -Wl,-soname,testpython.so >/dev/null 2>&1 && \
+    if ${CC} $PYTHON_CFLAGS -fPIC -shared -o testpython.so testpython.c -Wl,-soname,testpython.so $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
         ( objdump --headers --private-headers testpython.so | grep 'testpython' ) >/dev/null 2>&1; then
         result=yes
     else
--- origsrc/gnome-commander-1.4.1/libgcmd/Makefile.am	2014-03-13 16:13:19.000000000 -0500
+++ src/gnome-commander-1.4.1/libgcmd/Makefile.am	2014-04-30 20:41:13.435024200 -0500
@@ -5,7 +5,7 @@ AM_CPPFLAGS = \
 	$(GNOMEUI_CFLAGS) \
 	$(GNOMEVFS_CFLAGS)
 
-pkglib_LTLIBRARIES = libgcmd.la
+noinst_LTLIBRARIES = libgcmd.la
 
 libgcmd_la_SOURCES = \
 	gnome-cmd-plugin.h gnome-cmd-plugin.cc \
--- origsrc/gnome-commander-1.4.1/plugins/fileroller/Makefile.am	2014-03-13 16:13:19.000000000 -0500
+++ src/gnome-commander-1.4.1/plugins/fileroller/Makefile.am	2014-04-30 20:41:33.152028000 -0500
@@ -14,6 +14,6 @@ libfileroller_la_SOURCES = \
 	file-roller-plugin.h file-roller-plugin.cc \
 	file-roller.xpm file-roller-small.xpm
 
-libfileroller_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version
+libfileroller_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version $(NO_UNDEFINED)
 
 -include $(top_srcdir)/git.mk
--- origsrc/gnome-commander-1.4.1/plugins/test/Makefile.am	2014-03-13 16:13:19.000000000 -0500
+++ src/gnome-commander-1.4.1/plugins/test/Makefile.am	2014-04-30 20:41:43.754374300 -0500
@@ -14,6 +14,6 @@ libtest_la_SOURCES = \
 	test-plugin.h test-plugin.cc \
 	test-plugin.xpm
 
-libtest_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version
+libtest_la_LDFLAGS = $(GNOMEUI_LIBS) $(GNOMEVFS_LIBS) -module -avoid-version $(NO_UNDEFINED)
 
 -include $(top_srcdir)/git.mk
--- origsrc/gnome-commander-1.4.1/src/Makefile.am	2014-03-25 09:55:07.000000000 -0500
+++ src/gnome-commander-1.4.1/src/Makefile.am	2014-04-30 20:42:31.235403600 -0500
@@ -112,6 +112,11 @@ gnome_commander_LDADD = \
 	$(PYTHON_LIBS) \
 	$(PYTHON_EXTRA_LIBS)
 
+gnome_commander_LDFLAGS = -export-dynamic
+if PLATFORM_WIN32
+gnome_commander_LDFLAGS += -Wl,--out-implib,gnome-commander.exe.a
+endif
+
 gcmd_block_SOURCES = block.cc
 
 -include $(top_srcdir)/git.mk
